### PR TITLE
Marked classes and interfaces as API

### PR DESCRIPTION
**Pull request by ProkopovVitaliy**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#32054

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by ProkopovVitaliy** [comment]
@magetno run Static Tests

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by ProkopovVitaliy** [comment]
@magento run Database Compare, Semantic Version Checker

**Comment by ProkopovVitaliy** [comment]
@magento run Semantic Version Checker

File: lib/internal/Magento/Framework/App/Action/HttpGetActionInterface.php [hunk 1]
```diff
@@ -8,10 +8,14 @@
...
 
+use Magento\Framework\App\ActionInterface;
+
 /**
  * Marker for actions processing GET requests.
+ *
+ * @api
  */
-interface HttpGetActionInterface extends HttpHeadActionInterface
+interface HttpGetActionInterface extends ActionInterface
```
Comment (eduard13) [inline lib/internal/Magento/Framework/App/Action/HttpGetActionInterface.php]: Let's revert this change. It will be done in scope of #32328.

File: app/code/Magento/Widget/Block/BlockInterface.php [hunk 2]
```diff
@@ -8,6 +8,7 @@
  * Widget Block Interface
  *
  * @author      Magento Core Team <core@magentocommerce.com>
+ * @api
```
Comment (eduard13) [inline app/code/Magento/Widget/Block/BlockInterface.php:11]: Please remove the `@author` annotation.

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests B2B, Functional Tests CE, Unit Tests

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Database Compare, Functional Tests B2B

**Review by eduard13 — APPROVED** [review]
Don't think that Database Compare failed builds are somehow related with these changes.

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8921](https://jira.corp.magento.com/browse/ENGCOM-8921) has been created to process this Pull Request

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by ProkopovVitaliy** [comment]
@magento run Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Functional Tests B2B

**Comment by gabrieldagama** [comment]
Hi @ProkopovVitaliy can you recreate this PR to the `api-changes-develop` branch instead? Just make sure to create a new branch from this one and cherry-pick your commit.

Thanks!

**Comment by gabrieldagama** [comment]
Closed in favor of https://github.com/magento/magento2/pull/33221
